plugin/src/test/java/io/jenkins/plugins/remotingkafka/KafkaComputerLauncherTest.java

```diff
@@ -1,27 +1,32 @@
 package io.jenkins.plugins.remotingkafka;
 
-import com.salesforce.kafka.test.junit4.SharedKafkaTestResource;
 import hudson.model.Computer;
 import hudson.model.FreeStyleProject;
 import hudson.slaves.DumbSlave;
 import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
 import org.jvnet.hudson.test.JenkinsRule;
+import org.testcontainers.containers.DockerComposeContainer;
+
+import java.io.File;
 
 import static org.junit.Assert.assertNotNull;
 
 public class KafkaComputerLauncherTest {
     @ClassRule
-    public static final SharedKafkaTestResource sharedKafkaTestResource = new SharedKafkaTestResource();
+    public static DockerComposeContainer environment = new DockerComposeContainer(new File("src/test/resources/compose-test.yml"))
```

Comment:
Why not KafkaContainer from `org.testcontainers:kafka` module?

Reply:
@bsideup I had this error when using `org.testcontainers:kafka` with my plugin (to connect a consumer to)
```
Group coordinator tc-1qr9ZwtX:9092 (id: 2147483646 rack: null) is unavailable or invalid, will attempt rediscovery
```
The way I used to connect (using Kafka with embedded Zookeeper):
```
try (KafkaContainer kafka = new KafkaContainer()) {
            kafka.start();
            GlobalKafkaConfiguration g = GlobalKafkaConfiguration.get();
            String kafkaURL = kafka.getContainerIpAddress() + ":" + kafka.getMappedPort(9092);
            String zookeeperURL = kafka.getContainerIpAddress() + ":" + kafka.getMappedPort(2181);
            g.setBrokerURL(kafkaURL);
            g.setZookeeperURL(zookeeperURL);
```

Reply:
yes, because you should use `String kafkaURL = kafka.getBootstrapServers()` :)

Reply:
thanks @bsideup, it works in my local machine now, it should work here also :). Btw, why `getBootstrapServers()` and `kafka.getMappedPort(9092)` return different port? So kafka is not mapping to port 9092?

Reply:
Because Kafka doesn't work with NAT-like environments and we had to re-map it to make it work :)